### PR TITLE
implemented functionality for user authentication with Google 

### PR DESCRIPTION
**Pull request by Timothy-py**
Highlights

- Installed some app dependencies
- Added some environment variables
- Implemented passport utility
- Created Google Auth model
- Created Google Auth route
- Created Google Auth controller
- Updated model README file

### CODE REVIEW

File: model/googleAuth.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const googleAuthSchema = new mongoose.Schema({
+    google_id: {
+        type: Number,
+        required: true
+    },
+    display_name: {
+        type: String,
+        required: true
+    },
+    first_name: {
```
Comment (romzy234) [inline model/googleAuth.js:12]: # Google Oauth Returning without  First name and Last Name

There have been instances in which the user signup Google Service without providing their name or other details [Read This](https://stackoverflow.com/questions/58316184/get-empty-first-name-and-last-name-for-some-users-from-google-oauth) 

Did You Make a **Fail Use Case** For This?

Comment (romzy234) [reply model/googleAuth.js:12]: This Fields Are Kept As Required

File: controllers/google.auth.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * @author Timothy Adeyeye <adeyeyetimothy33@gmail.com>
+ * @description Authenticaton with Google OAuth
+ * @route `/api/v1/auth/register`
+ * @access Public
+ * @type GET
+ */
+exports.profile = (req, res) => {
+    const user = req.user
```
Comment (romzy234) [inline controllers/google.auth.js:17]: Did Passport Write `req.user` this to memory

File: package.json [hunk 3]
```diff
@@ -22,16 +22,20 @@
     "dotenv": "^16.0.2",
     "ejs": "^3.1.8",
     "express": "~4.16.1",
+    "express-session": "^1.17.3",
     "google-auth-library": "^8.6.0",
     "helmet": "^6.0.0",
     "http-errors": "~1.6.3",
     "joi": "^17.6.4",
     "jsonwebtoken": "^8.5.1",
+    "memorystore": "^1.6.7",
```
Comment (romzy234) [inline package.json:31]: Why Install Memorystore ?

**Review by romzy234 — CHANGES_REQUESTED** [review]
# Overview for issue #12 

So The Code Is Okay But 

You are using `Passport Oauth Strategy` not just for the web callback but for the whole auth process, which you later rewrite using jwt 

it works but not the best way to solve this 

I assume you found it quite challenging to use the SDK 

So I would Like to create a Staging branch and work on that incorporation together with you

File: model/googleAuth.js (hunk 1 above)
Comment (Timothy-py) [reply model/googleAuth.js:12]: Oh, I never knew that it's possible for Google service not to return some of those fields. Okay I'll make appropriate changes to that.

File: controllers/google.auth.js (hunk 2 above)
Comment (Timothy-py) [reply controllers/google.auth.js:17]: serialize

Yes, I think from the serialize method

File: package.json (hunk 3 above)
Comment (Timothy-py) [reply package.json:31]: session

From express-session middleware documentation, the default MemoryStore is not fit for a production environment.

memorystore

There are other recommended compatible modules for session store but the memorystore module pretty much works well also for a production use case.

**Comment by Timothy-py** [comment]
Alright, thanks.

**Pull request closed without merging**